fabric/src/main/java/com/gregtechceu/gtceu/common/blockentity/fabric/FluidPipeBlockEntityImpl.java

```diff
@@ -39,7 +39,7 @@ public static FluidPipeBlockEntity create(BlockEntityType<?> type, BlockPos pos,
     public static void onBlockEntityRegister(BlockEntityType<FluidPipeBlockEntity> type) {
         FluidStorage.SIDED.registerForBlockEntity((blockEntity, direction) -> {
             final var fluidTransfer = blockEntity.getFluidHandler(direction);
-            return fluidTransfer == null ? null : new Storage<FluidVariant>() {
+            return fluidTransfer == null ? null : new Storage<>() {
```

Comment:
Im not sure whether you have tried fabric, but it will crash if you do it.

Reply:
It's not supposed to. ^^

Reply:
Damn, it doesnt even compile :eyes: Will fix :)

Reply:
I moved the anonymous implementations over to some inner records, the game starts up fine now. 
By crash did you mean crash at client initialization or for a specific case? If it is a specific case, how should i test this?

Reply:
yep, I meaned it doesnt compile, cus I tried the same thing before, lol